packages/@aws-cdk/service-spec-importers/src/types/registry-schema/JsonSchema.ts

```diff
@@ -199,6 +206,47 @@ export namespace jsonschema {
     return !(x as RecordLikeObject).properties;
   }
 
+  /**
+   * Relationship as defined in the CloudFormation Registry Schema.
+   * Represents how resource property relationships are specified in the source schema files.
+   */
+  export interface RelationshipRefSchema {
+    /** The cloudFormationType (e.g. 'AWS::S3::Bucket') */
+    readonly typeName: string;
+    /** The property path (e.g. '/properties/BucketName') */
+    readonly propertyPath: string;
+  }
+
+  /**
+   * Determines if a schema is a relationshipRef
+   */
+  function isRelationshipRef(x: any): x is RelationshipRefSchema {
+    return (
+      x &&
+      'typeName' in x &&
+      typeof x.typeName === 'string' &&
+      'propertyPath' in x &&
+      typeof x.propertyPath === 'string'
+    );
+  }
+
+  /**
+   * Determines if a schema contains a relationshipRef
+   * This function handles the two following cases:
+   * { type: 'string', relationshipRef: {...} } -> occurs when there a single relationship for a property
+   * { relationshipRef: {...} } -> occurs when there are multiple relationships as the source data looks like this:
+   * { type: 'string', anyOf: [ { relationshipRef: {...} }, ...]} (the type is not present along relationshipRef)
+   */
+  export function containsRelationship(x: any): x is { relationshipRef: RelationshipRefSchema } {
```

Comment:
Given the types below, the code that needs this should be able to do something like:

```ts
const relationship = isString(x) ? x.relationshipRef : undefined;
```

Don't think we need the type testing on arbitrary objects to see if they are `RelationshipRef`s?